gridcomps/History/Sampler/MAPL_StationSamplerMod.F90

```diff
@@ -237,93 +336,146 @@ subroutine add_metadata_route_handle (this,bundle,timeInfo,vdata,rc)
-       call v%add_attribute('missing_value', MAPL_UNDEF)
-       call v%add_attribute('_FillValue',    MAPL_UNDEF)
-       call v%add_attribute('valid_range',   (/-MAPL_UNDEF,MAPL_UNDEF/))
-       call this%fmd%add_variable(trim(var_name),v,_RC)
-    end do
-    deallocate (fieldNameList)
+       call iter%next()
+    enddo
 
 
-    !__ 3. locstream route handle
+    !__ 3. route handle  CS --> LS_ds
     !
     call ESMF_FieldBundleGet(bundle,grid=grid,_RC)
-    this%regridder = LocStreamRegridder(grid,this%esmf_ls,_RC)
+    this%regridder = LocStreamRegridder(grid,this%LS_ds,_RC)
 
+    !__ 4. route handle  LS_ds --> LS_chunk
+    !
+    src_field = ESMF_FieldCreate(this%LS_ds,typekind=ESMF_TYPEKIND_R4,gridToFieldMap=[1],_RC)
+    chunk_field = ESMF_FieldCreate(this%LS_chunk,typekind=ESMF_TYPEKIND_R4,gridToFieldMap=[1],_RC)
+    call ESMF_FieldGet( src_field, localDE=0, farrayPtr=pt1, _RC )
+    call ESMF_FieldGet( chunk_field, localDE=0, farrayPtr=pt2, _RC )
+    pt1=0.0
+    pt2=0.0
+    call ESMF_FieldRedistStore(src_field,chunk_field,this%RH,_RC)
+    call ESMF_FieldDestroy(src_field,noGarbage=.true.,_RC)
+    call ESMF_FieldDestroy(chunk_field,noGarbage=.true.,_RC)
 
     _RETURN(_SUCCESS)
   end subroutine add_metadata_route_handle
 
 
+  subroutine  create_metadata_variable(this,vname,rc)
+    class(StationSampler), intent(inout) :: this
+    character(len=*), intent(in)         :: vname
+    integer, optional, intent(out)       :: rc
+
+    type(ESMF_Field) :: field
+    type(variable) :: v
+    logical :: is_present
+    integer :: field_rank, status
+    character(len=ESMF_MAXSTR) :: var_name,long_name,units,vdims
+    integer :: rank,lb(1),ub(1)
+    integer :: k, ig
+
+
+    call ESMF_FieldBundleGet(this%bundle,vname,field=field,_RC)
+    call ESMF_FieldGet(field,name=var_name,rank=field_rank,_RC)
+    call ESMF_AttributeGet(field,name="LONG_NAME",isPresent=is_present,_RC)
+    if ( is_present ) then
+       call ESMF_AttributeGet  (FIELD, NAME="LONG_NAME",VALUE=long_name, _RC)
+    else
+       long_name = var_name
+    endif
+    call ESMF_AttributeGet(field,name="UNITS",isPresent=is_present,_RC)
+    if ( is_present ) then
+       call ESMF_AttributeGet  (FIELD, NAME="UNITS",VALUE=units, _RC)
+    else
+       units = 'unknown'
+    endif
+
+!    -- in future, replace keyword station by index_name_x as in trajectory sampler
+!    if (field_rank==2) then
+!       vdims = this%index_name_x
+!    else if (field_rank==3) then
+!       vdims = trim(this%index_name_x)//",lev"
+!    end if
+
+    if (field_rank==2) then
+       vdims = "station_index,time"
+       v = variable(type=PFIO_REAL32,dimensions=trim(vdims),chunksizes=[this%nstation,1])
+    else if (field_rank==3) then
+       vdims = "lev,station_index,time"
+       call ESMF_FieldGet(field,ungriddedLBound=lb,ungriddedUBound=ub,_RC)
+       v = variable(type=PFIO_REAL32,dimensions=trim(vdims),chunksizes=[ub(1)-lb(1)+1,1,1])
+    end if
```

Comment:
Declare allocatable integer array chunksizes up above (if still needed) and then:
```suggestion
    vdims = "station_index,time"
    select case (field_rank)
    case (2)
       chunksizes = [this%nstation,1]
       v = variable(type=PFIO_REAL32,dimensions=trim(vdims),chunksizes=[this%nstation,1])
    case(3)
       vdims = "lev," // vdims
       call ESMF_FieldGet(field,ungriddedLBound=lb,ungriddedUBound=ub,_RC)
       chunksizes = [ub(1)-lb(1)+1,1,1]
    case default
       _FAIL('unsupported rank')
    end if
    v = variable(type=PFIO_REAL32,dimensions=trim(vdims),chunksizes=[ub(1)-lb(1)+1,1,1])
```

Reply:
I see what you mean; thanks and I have made the changes.